In [7]:
import pandas as pd

# Работа с Pandas

1. Загрузите ваш DataFrame в переменную df и выведете на экран первые пять строк 

In [8]:
path = r'G:\Мой диск\Colab Notebooks\1T\ДЗ_1T\2\googleplaystore.csv'# Считываем наш датафрейм из csv файла
df = pd.read_csv(path, encoding='utf-8') 

2. Отобразите количество строк и столбцов в DataFrame

In [9]:
df.shape

(10841, 13)

3. Выведете основную информацию о DataFrame методом info(). Проанализируйте есть ли пропуски в столбцах. Напишите вывод о том, какие столбцы содержат пропуски

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  float64
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(2), object(11)
memory usage: 1.1+ MB


В ДФ - 13 столбцов. и 10841 строк. В ДФ есть пропуски в столбцах: Rating, Type, Content Rating, Current Ver, Android Ver

4. Удалите или замените пропуски известными вам методами. (Подсказка: `dropna()` или `fillna()`, выбор того или иного метода должен быть аргументирован).

In [11]:
df.fillna(0, axis=1, inplace=True)

`dropna()` применяется тогда, когда нам надо удалить строки или столбцы с пропущенными значениями ЦЕЛИКОМ. В данном случае, это потеря данных, для нас, это не приемлимо.
Применил метод .fillna, заменены пропуски (NaN) на 0.

5. Проверьте нет ли пропусков после произведенной обработки

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   App             10841 non-null  object
 1   Category        10841 non-null  object
 2   Rating          10841 non-null  object
 3   Reviews         10841 non-null  object
 4   Size            10841 non-null  object
 5   Installs        10841 non-null  object
 6   Type            10841 non-null  object
 7   Price           10841 non-null  object
 8   Content Rating  10841 non-null  object
 9   Genres          10841 non-null  object
 10  Last Updated    10841 non-null  object
 11  Current Ver     10841 non-null  object
 12  Android Ver     10841 non-null  object
dtypes: object(13)
memory usage: 1.1+ MB


                            ПРОПУСКОВ НЕТ

6. Выведите на экран минимум и максимум из столбца `Price`:

In [13]:
min_price = df['Price'].min()
max_price = df['Price'].max()
print('минимальное значение: ', min_price)
print('максимальное значение: ', max_price)

минимальное значение:  0.0
максимальное значение:  400.0


7. Выведите на экран медиану и среднее арифметическое столбцов:
 - Rating
 - Reviews

In [14]:
print(f"медианное значение столбца Rating: {df['Rating'].median()}")
print(f"среднее арифметическое значение столбца Rating: {df['Rating'].mean()}")

медианное значение столбца Rating: 4.2
среднее арифметическое значение столбца Rating: 3.6231897426436723


    Колонка Reviews требовала перевода формата в числовой  !!

In [15]:
df['Reviews'] = pd.to_numeric(df['Reviews'], errors='coerce') # переводим в колонку в числовой формат

print(f'медианное значение столбца Reviews: {df["Reviews"].median()}')
print(f'среднее арифметическое значение столбца Reviews: {df["Reviews"].mean()}')

медианное значение столбца Reviews: 2094.0
среднее арифметическое значение столбца Reviews: 444152.89603321033


8. Выведите на экран все уникальные значения категориального столбца 'Genres'

In [16]:
summ_uniq = df['Genres'].describe(include='object').unique() # через .unique
print('всего уникальных значений:',summ_uniq[1], 'единиц')


print(f"всего уникальных значений: {str(df['Genres'].nunique())} единиц")   
# через .nunique обернуто в str(), для преобразования в строковый тип данных + f строка

всего уникальных значений: 120 единиц
всего уникальных значений: 120 единиц


In [17]:
print(f"{df['Genres'].unique()}")

['Art & Design' 'Art & Design;Pretend Play' 'Art & Design;Creativity'
 'Art & Design;Action & Adventure' 'Auto & Vehicles' 'Beauty'
 'Books & Reference' 'Business' 'Comics' 'Comics;Creativity'
 'Communication' 'Dating' 'Education;Education' 'Education'
 'Education;Creativity' 'Education;Music & Video'
 'Education;Action & Adventure' 'Education;Pretend Play'
 'Education;Brain Games' 'Entertainment' 'Entertainment;Music & Video'
 'Entertainment;Brain Games' 'Entertainment;Creativity' 'Events' 'Finance'
 'Food & Drink' 'Health & Fitness' 'House & Home' 'Libraries & Demo'
 'Lifestyle' 'Lifestyle;Pretend Play' 'Adventure;Action & Adventure'
 'Arcade' 'Casual' 'Card' 'Casual;Pretend Play' 'Action' 'Strategy'
 'Puzzle' 'Sports' 'Music' 'Word' 'Racing' 'Casual;Creativity'
 'Casual;Action & Adventure' 'Simulation' 'Adventure' 'Board' 'Trivia'
 'Role Playing' 'Simulation;Education' 'Action;Action & Adventure'
 'Casual;Brain Games' 'Simulation;Action & Adventure'
 'Educational;Creativity' 'Puzzle

    Вывод перечня уникальных названий

9. Сгруппируйте данные по столбцу `Genres` и посчитайте для каждого жанра средний и медианный рейтинг. Представьте результат в виде нового DataFrame,который будет сохранен в переменную `grouped_df`, где одна колонка будет содержать названия жарнов, а две другие буду содержать в себе средний и медианный рейтинги. Напишите краткий вывод.

In [25]:
grouped_data = df.groupby('Genres') # Группировка данных по столбцу 'Genres' 
mean_rating = grouped_data['Rating'].mean()      # Вычисляем средний рейтинг
median_rating = grouped_data['Rating'].median()  # Вычисляем медианный рейтинг

# Создаем новый DataFrame с результатами
grouped_df = pd.DataFrame({'Названия жанров': mean_rating.index, 'Средний рейтинг': mean_rating.values, 'Медианный рейтинг': median_rating.values})

# Вывод нового DataFrame grouped_df
print(f'{grouped_df.sort_values(by="Медианный рейтинг", ascending=False)}')

                     Названия жанров Средний рейтинг Медианный рейтинг
60                 February 11, 2018            19.0              19.0
35                 Comics;Creativity             4.8               4.8
18                Board;Pretend Play             4.8               4.8
65        Health & Fitness;Education             4.7               4.7
11           Art & Design;Creativity             4.4               4.7
..                               ...             ...               ...
79               Parenting;Education        1.657143               0.0
10   Art & Design;Action & Adventure             0.0               0.0
114                 Trivia;Education             0.0               0.0
95            Role Playing;Education             0.0               0.0
20      Books & Reference;Creativity             0.0               0.0

[120 rows x 3 columns]


    По медианному и среднему рейтингу - самый топовый жанр - это "February 11, 2018" что бы это нибыло ))
    Вероятно в данных рейтингов ошибка, так как на лицо слишком сильное расхождение в значениях 1 и 2ого места.


10. Из получившегося датафрейма выведите на экран только те жанры, чей медианный рейтинг больше 4.5. Напишите краткий вывод.

In [29]:
grouped_df.sort_values(by="Медианный рейтинг", ascending=False).loc[(grouped_df['Медианный рейтинг'] > 4.5 ),['Названия жанров', 'Медианный рейтинг']]

Названия жанров Медианный рейтинг
60             February 11, 2018              19.0
35             Comics;Creativity               4.8
18            Board;Pretend Play               4.8
65    Health & Fitness;Education               4.7
11       Art & Design;Creativity               4.7
88              Puzzle;Education               4.6
4          Adventure;Brain Games               4.6
75           Music;Music & Video               4.6
55      Entertainment;Creativity               4.6
106  Strategy;Action & Adventure               4.6

    Выводы: в строке 60 - выброс. Так как вероятнее всего максимальный рейтинг - 5 + название жинра - НЕ Жанр. Строку надо ДРОПНУТЬ! 

11. Выведете кол-во дубликатов в основном DataFrame (df) и удалите их 

In [34]:
print(f'Количество дубликатов в основном DataFrame = {df.duplicated().sum()}')

Количество дупликатов в основном DataFrame = 483


In [39]:
df.drop_duplicates(inplace = True)

In [40]:
print(f'Количество дупликатов в основном DataFrame = {df.duplicated().sum()}')

Количество дупликатов в основном DataFrame = 0


<b> Было 483 дубликата - внес изменения в датасет inplace = True

12. Посчитатайте какое количество приложений содержится в каждом жанре (value_counts).В получившемся Series выведите на экран 10 строк с наибольшим количеством приложений отсортированных по убыванию.

In [41]:
# Считаем количество приложений в каждом жанре
app_count = df['Genres'].value_counts()

# Выводим 10 строк с наибольшим количеством приложений, отсортированных по убыванию
top_10_genres = app_count.head(10)
print(top_10_genres)


Genres
Tools              842
Entertainment      588
Education          527
Business           427
Medical            408
Productivity       407
Personalization    388
Lifestyle          372
Communication      366
Sports             364
Name: count, dtype: int64
